### PR TITLE
Feat / Better url formatting

### PR DESCRIPTION
**Pull request by royschut**
## Better URL formatting
I had refactored url formatting, but we still found that the `createURL` utility was having too many roles mixed up. Then I had the idea to simply split the location based url creation, because these are the only occasions where we need to take out the search string and put it back in (which made it weird, being mixed up with query params). It makes a lot of sense to me to leave that part to the `location` util in `ui-react` and let `urlFomatting` util in `common` do the actual parsing using URLSearchParams.

I also added unit tests, to make sure all cases work well.

### CODE REVIEW

File: packages/ui-react/src/pages/ScreenRouting/playlistScreens/PlaylistLiveChannels/PlaylistLiveChannels.tsx [hunk 1]
```diff
@@ -134,6 +134,7 @@ const PlaylistLiveChannels: ScreenComponent<Playlist> = ({ data: { feedid, playl
     <ShareButton title={channelMediaItem.title} description={channelMediaItem.description} url={window.location.href} />
   ) : null;
 
+  // @todo
```
Comment (ChristiaanScheermeijer) [inline packages/ui-react/src/pages/ScreenRouting/playlistScreens/PlaylistLiveChannels/PlaylistLiveChannels.tsx]: todo?

Comment (royschut) [reply packages/ui-react/src/pages/ScreenRouting/playlistScreens/PlaylistLiveChannels/PlaylistLiveChannels.tsx]: Ah yes, I wanted to fix the createURL + liveChannelsURL combination, by giving the latter more args/options, but maybe from the routes refactor. I will remove the todo.

**Review by ChristiaanScheermeijer — APPROVED** [review]
Nice cleanup @royschut! Does this also work correctly for the absolute `redirectUrl` properties?

File: packages/common/src/utils/urlFormatting.test.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  test('valid url from a path, query params', async () => {
+    const url = createURL('/test', { foo: 'bar' });
+
+    expect(url).toEqual('/test?foo=bar');
+  });
+  test('valid url from a path including params, query params', async () => {
+    const url = createURL('/test?existing-param=1', { foo: 'bar' });
+
+    expect(url).toEqual('/test?existing-param=1&foo=bar');
+  });
```
Comment (langemike) [inline packages/common/src/utils/urlFormatting.test.ts:13]: Great that you also wrote some tests for it. Maybe also add a test for the removal from a queryParam?

```
 test('valid url from a path including params, query params', async () => {
    const url = createURL('/test?foo=1&bar=1', { foo: null});

    expect(url).toEqual('/test?bar=1');
  });
```

**Review by langemike — APPROVED** [review]
Nice approach! I have only one suggestion for a `createURL` test. But I consider this non-blocking ;)
Also you wrote you are going to remove the `//todo`

File: packages/common/src/utils/urlFormatting.test.ts (hunk 2 above)
Comment (royschut) [reply packages/common/src/utils/urlFormatting.test.ts:13]: Good idea! Added it.

**Comment by royschut** [comment]
> Nice cleanup @royschut! Does this also work correctly for the absolute `redirectUrl` properties?

Yes! And I'm proving it with another unit test. :-)

**Pull request merged**